src/Symfony/Component/Security/Core/Authorization/IsGrantedPayload.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Core\Authorization;
+
+use Symfony\Component\Security\Core\Authentication\AuthenticationTrustResolverInterface;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Symfony\Component\Security\Core\User\UserInterface;
+
+/**
+ * @template T
+ *
+ * @author Alexandre Daubois <alex.daubois@gmail.com>
+ */
+final class IsGrantedPayload
```

Comment:
```suggestion
final readonly class IsGrantedPayload
```

Given that the same payload might be passed to multiple closures, we might want to make it immutable.